controller/codebase.go

```diff
@@ -118,6 +118,24 @@ func (c *CodebaseController) Edit(ctx *app.EditCodebaseContext) error {
 	return ctx.OK(resp)
 }
 
+func (c *CodebaseController) Delete(ctx *app.DeleteCodebaseContext) error {
+	_, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
```

Comment:
Missing some auth check to verify you're allowed to delete the codebase. @alexeykazakov I assume only space-owner should be allowed at this point? Or collaborators?

Reply:
It should be the space-owner only for now. We will improve it later when we have a new authorization service ready.
@nuralivirani see an example of space owner check here:
https://github.com/fabric8-services/fabric8-wit/blob/master/controller/space_codebases.go#L27-L30
https://github.com/fabric8-services/fabric8-wit/blob/master/controller/space_codebases.go#L49-L51

Reply:
@alexeykazakov thanks :-)

@aslakknutsen, @alexeykazakov I think for this check
first we will load codebase for given ctx.codebaseID
then we will load space for codebase.SpaceID
then we will check if space.OwnerID == *identityID

We need to load codebase and space as we only get codebaseID in request for Codebase Removal API.  Let me know in case any mis-understanding.  I will make this changes.

Reply:
Currently, there is some mess in /codebase/:codeBaseID and /spaces/:spaceID/codbases/:codeBaseID endpoints. 
/codebase/:codeBaseID currently has Show action which shows the codebase. The rest actions like Create and Edit actually creates and edits Che Workspaces. Not codebases.
/spaces/:spaceID/codbases/:codeBaseID has Creates (creates a codebase) and LIst (list codebases)
So, maybe we should add DELETE to /spaces/:spaceID/codbases/:codeBaseID instead of /codebase/:codeBaseID? Then we will both Create and Delete actions in the same endpoint which sounds better to me. It also will be easier to check the space owner.
And we IMO should rename workspace related endpoints/actions to something more clear ;) In a separate PR though.

Reply:
@alexeykazakov First, previously me and @aslakknutsen had discussion on the same.  At that time, aslak has suggested not to use SpaceID for codebase DELETE.
Second, after your arguments I spend some more time to understand different entity and here is what I observed.  For Workitem entity, its CREATE takes SpaceID but DELETE only takes WorkItemID.  For WorkItemLinkType entity, its CREATE takes SpaceID and DELETE also takes SpaceID along with WorkItemTypeLinkID.
Third, primary key for all these entities (codebase, Workitem, WorkItemTypeLink) is only id field.  If you consider only third point then it's sufficient to delete these entity with only Id param.

@aslakknutsen and @alexeykazakov plz help here to finalize the design decision after that I will make required changes.

Reply:
The problem with codebase endpoint is that it also manages workspaces. And it’s quite confusing to have POST /codebase to create a workspace and DELETE /codebase to delete a codebase.
But I agree it makes sense to keep the codebase delete action in the /codebase endpoint and move the workspace related ones somewhere else (in different PR).

Reply:
@alexeykazakov That's not how it works.. :)

POST spaces/id/codebases - creates a codebase for a given space
GET spaces/id/codebases - list codebases for a given space

GET codebases/id - get a single codebase
GET codebase/id/edit - trigger edit flow of a workspace, response is existing workspaces and a link to create
POST codebase/id/create - create a workspace
POST codebase/id/open - opens an existing workspace

Reply:
Ah, right. But anyway, codebase/id/edit, codebase/id/create, codebase/id/open IMO are confusing names for endpoints managing workspaces ;) There is even more confusion in the goa controllers/action names.

Reply:
Well, the point was to hide Workspaces.

So edit(open|create) a codebase sound perfectly normal to represent an IDE action :)